osu.Game/Screens/Select/Carousel/TopLocalRank.cs

```diff
@@ -30,72 +30,45 @@ public class TopLocalRank : UpdateableRank
         [Resolved]
         private IAPIProvider api { get; set; }
 
+        private IDisposable scoreSubscription;
+
+        private bool rankUpdatePending;
+
         public TopLocalRank(BeatmapInfo beatmapInfo)
             : base(null)
         {
             this.beatmapInfo = beatmapInfo;
         }
 
-        [BackgroundDependencyLoader]
-        private void load()
-        {
-            ruleset.ValueChanged += _ => fetchAndLoadTopScore();
-
-            fetchAndLoadTopScore();
-        }
-
         protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            scoreSubscription = realmFactory.Context.All<ScoreInfo>()
-                                            .Filter($"{nameof(ScoreInfo.BeatmapInfo)}.{nameof(BeatmapInfo.ID)} = $0", beatmapInfo.ID)
-                                            .QueryAsyncWithNotifications((_, changes, ___) =>
-                                            {
-                                                if (changes == null)
-                                                    return;
-
-                                                fetchTopScoreRank();
-                                            });
-        }
-
-        private IDisposable scoreSubscription;
-
-        private ScheduledDelegate scheduledRankUpdate;
-
-        private void fetchAndLoadTopScore()
-        {
-            // TODO: this lookup likely isn't required, we can use the results of the subscription directly.
-            var rank = fetchTopScoreRank();
-
-            scheduledRankUpdate = Scheduler.Add(() =>
+            ruleset.BindValueChanged(_ =>
             {
-                Rank = rank;
-
+                rankUpdatePending = true;
                 // Required since presence is changed via IsPresent override
                 Invalidate(Invalidation.Presence);
```

Comment:
This invalidation needs to be inside of the subscription callback, because otherwise removing the last user score on a map leaves a gap on the beatmap panel:

![2022-01-18-211750_392x76_scrot](https://user-images.githubusercontent.com/20418176/150012197-e9b371a0-0443-444c-9859-604d7affb424.png)

Reply:
I saw this during testing but didn't think much of it (thought it was the case on master). Didn't put one and one together with the invalidate calls.